src/activity.ts

```diff
@@ -72,3 +87,47 @@ export function handleSwapActivityEntity(event: SwapEvent): void {
 
   activity.save();
 }
+
+export function handleAddCollateralActivityEntity(
+  event: AddCollateralEvent,
+  controllerId: string
+): void {
+  const token = loadOrCreateERC721Token(event.params.collateralAddress);
+  if (!token) return;
+
+  let activity = Activity.load(event.transaction.hash.toHex());
+  if (!activity) {
+    activity = new Activity(event.transaction.hash.toHex());
+    activity.type = "ADD_COLLATERAL";
+    activity.timestamp = event.block.timestamp.toI32();
+    activity.controller = controllerId;
+    activity.user = event.transaction.from;
+    activity.vault = generateVaultId(
+      event.params._event.address,
+      event.params.account,
+      token
+    );
+  }
+
+  const activityAddedCollateral = new ActivityAddedCollateral(
+    generateActivityCollateralId(
```

Comment:
don't we want the tx hash to always be the ID? And I thought we just had one new `Activity` type?

Reply:
id of the added collateral is the activity tx hash + collateral